src/wilcoxon.jl

```diff
@@ -163,6 +163,27 @@ function StatsAPI.pvalue(x::ExactSignedRankTest; tail=:both)
     end
 end
 
+function psignrank(W, n, lower_tail)
+    int_type = typeof(n)
+    W = int_type(W)
+    two = one(int_type) + one(int_type)
+    max_W =  n * (n + one(int_type)) ÷ two + one(int_type)
+    DP = zeros(int_type(n), max_W)
+    DP[begin] = one(int_type)
```

Comment:
Since `DP` is a `Vector`, you could just use
```suggestion
    DP[1] = one(int_type)
```